middleman_protocol/middleman_protocol/message.py

```diff
@@ -0,0 +1,196 @@
+import sys
+from hashlib import sha256
+
+from construct import Byte
+from construct import Bytes
+from construct import Const
+from construct import Container
+from construct import Enum
+from construct import GreedyBytes
+from construct import Int16ub
+from construct import Int32ub
+from construct import Prefixed
+from construct import Struct
+from construct import VarInt
+
+from golem_messages.message.base import Message as BaseGolemMessage
+
+from .constants import FRAME_SEPARATOR
+from .constants import PayloadType
+from .constants import PAYLOAD_TYPE_TO_MIDDLEMAN_MESSAGE_CLASS
+from .exceptions import MiddlemanProtocolError
+
+
+class AbstractMiddlemanMessage:
+
+    __slots__ = [
+        'payload',
+        'payload_type',
+        'request_id',
+    ]
+
+    def __init__(self, payload, request_id):
+        if type(self) == AbstractMiddlemanMessage:
+            raise NotImplementedError(
+                'AbstractMiddlemanMessage cannot be instantiated directly.'
+            )
```

Comment:
Wouldn't [abc](https://docs.python.org/3/library/abc.html) do that for you?

Reply:
:+1: done